src/main/java/Entity/User.java

```diff
@@ -0,0 +1,93 @@
+package Entity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class User {
+    final int uid;
+    String username;
```

Comment:
Wouldn't first name/last name be needed in a company setting? maybe in userfile. Also why have username and uid? You can't have duplicate usernames or else you have to login with uid?

Reply:
Sure, first name/last name make sense! We could add it. Anyway, is put it into User abstract class a good idea? Maybe is better to be put into a common User(which we have deleted, but we can add back if needed)? id is given by system and could never be changed, but username can be customised by users. Users can log in through its username or its id with correct password. So I prefer to keep it. Both username and id need to be unique. What do you think? Moreover, what kind of type for first/last name is better? Do you think this is good: two string attributes, and one for first name one for last name?

Reply:
I delete username in User and keep username in UserFile. What do you think?

Reply:
Makes sense for username, and strings for first + last name sounds good. I guess username and id makes sense like uoft.